sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,177 @@
+	}, nil
+}
+
+// GetToken obtains a token from Azure Active Directory, using the Azure CLI command to authenticate.
+// ctx: Context used to control the request lifetime.
+// opts: TokenRequestOptions contains the list of scopes for which the token will have access.
+// Returns an AccessToken which can be used to authenticate service client calls.
+func (c *AzureCLICredential) GetToken(ctx context.Context, opts azcore.TokenRequestOptions) (*azcore.AccessToken, error) {
+	// The following code will remove the /.default suffix from the scope passed into the method since AzureCLI expect a resource string instead of a scope string
+	if strings.HasSuffix(opts.Scopes[0], defaultSuffix) {
+		opts.Scopes[0] = opts.Scopes[0][:len(opts.Scopes[0])-len(defaultSuffix)]
+	}
+	return c.authenticate(ctx, opts.Scopes[0])
+}
+
+// AuthenticationPolicy implements the azcore.Credential interface on AzureCLICredential and calls the Bearer Token policy
+// to get the bearer token.
+func (c *AzureCLICredential) AuthenticationPolicy(options azcore.AuthenticationPolicyOptions) azcore.Policy {
+	return newBearerTokenPolicy(c, options)
+}
+
+// TODO: should the CLI request have a timeout? This is currently not being used
+const timeoutCLIRequest = 10000
+
+// authenticate creates a client secret authentication request and returns the resulting Access Token or
+// an error in case of authentication failure.
+// ctx: The current request context
+// scopes: The scopes for which the token has access
+func (c *AzureCLICredential) authenticate(ctx context.Context, resource string) (*azcore.AccessToken, error) {
+	output, err := c.tokenProvider.GetCLIToken(ctx, resource)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.createAccessToken(output)
+}
+
+func (c *azureCLITokenProvider) GetCLIToken(ctx context.Context, resource string) ([]byte, error) {
+	// This is the path that a developer can set to tell this class what the install path for Azure CLI is.
+	const azureCLIPath = "AZURE_CLI_PATH"
+
+	// The default install paths are used to find Azure CLI. This is for security, so that any path in the calling program's Path environment is not used to execute Azure CLI.
+	azureCLIDefaultPathWindows := fmt.Sprintf("%s\\Microsoft SDKs\\Azure\\CLI2\\wbin; %s\\Microsoft SDKs\\Azure\\CLI2\\wbin", os.Getenv("ProgramFiles(x86)"), os.Getenv("ProgramFiles"))
+
+	// Default path for non-Windows.
+	const azureCLIDefaultPath = "/bin:/sbin:/usr/bin:/usr/local/bin"
+
+	// Validate resource, since it gets sent as a command line argument to Azure CLI
+	const invalidResourceErrorTemplate = "Resource %s is not in expected format. Only alphanumeric characters, [dot], [colon], [hyphen], and [forward slash] are allowed."
+	match, err := regexp.MatchString("^[0-9a-zA-Z-.:/]+$", resource)
+	if err != nil {
+		return nil, err
+	}
+	if !match {
+		return nil, fmt.Errorf(invalidResourceErrorTemplate, resource)
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, timeoutCLIRequest*time.Second)
```

Comment:
I don't see the `ctx` actually being used to control the lifetime of the request.

Reply:
Yeah, it isn't being used. I want to know if we do have to implement a timeout for this. Wondering what @schaabs thinks about whether we need the timeout? If we do then I can add the context to the command that will be executed.

Reply:
Well, since this is doing I/O then really IMO it should, especially since the call to `Output()` will wait for the invoked process to exit.

I don't see a variant of `Output()` that takes a context, looks like you'll have to roll your own.

Reply:
It works with CommandContext. Done.